### PR TITLE
fix a typo in ruby type of mount resource's supports

### PR DESCRIPTION
**Pull request by t2y**
See https://github.com/chef/chef/blob/master/lib/chef/resource/mount.rb#L45

### CODE REVIEW

File: chef_master/source/resource_mount.rst [hunk 1]
```diff
@@ -239,9 +239,9 @@ This resource has the following properties:
    .. end_tag
 
 ``supports``
-   **Ruby Type:** Array
+   **Ruby Type:** Hash
 
-   An array of options for supported mount features. Default value: ``:remount``.
+   Specify a Hash of supported mount features. Default value: ``remount: true``.
```
Comment (burtlo) [inline chef_master/source/resource_mount.rst]: Looking at the code and the comment above the default value, when not specifying a Hash of values, is `remount: false`.

File: chef_master/source/resource_mount.rst [hunk 2]
```diff
@@ -41,7 +41,7 @@ The full syntax for all of the properties that are available to the **mount** re
      password                   String
      provider                   Chef::Provider::Mount
      subscribes                 # see description
-     supports                   Array
+     supports                   Hash # defaults to { :remount => false }
```
Comment (burtlo) [inline chef_master/source/resource_mount.rst]: The interesting thing about the code is that `supports` does in fact support an `Array` as well. When an `Array` is specified the resulting `Hash` are all of the keys defined in the `Array` with values set to `true`.

Comment (burtlo) [reply chef_master/source/resource_mount.rst]: This mount resource using a `Hash`:
```
mount '/tmp' do
  supports { remount: true }
end
```
is equivalent to using an `Array`
```
mount '/tmp' do
  supports [ :remount ]
end
```

Comment (t2y) [reply chef_master/source/resource_mount.rst]: Hmm, it's true, but it's not appropriate.

File: chef_master/source/resource_mount.rst (hunk 1 above)
Comment (t2y) [reply chef_master/source/resource_mount.rst]: Oh, I fixed.

File: chef_master/source/resource_mount.rst (hunk 2 above)
Comment (burtlo) [reply chef_master/source/resource_mount.rst]: I wanted to add some comments in here around the code based on a conversation I was having someone at Chef. Hash all you want. Array all you want. 😄 I like seeing how code works and talk about it.

**Comment by burtlo** [comment]
I think this is good change to the docs. 👍 

Also, it took me a moment: the link that you provided above now points to a different line of code. Did you mean: https://github.com/chef/chef/blob/987bcb5dd740637693dbb6723e8730f32da43696/lib/chef/resource/mount.rb#L34-L35?

**Comment by t2y** [comment]
> Also, it took me a moment: the link that you provided above now points to a different line of code. Did you mean: https://github.com/chef/chef/blob/987bcb5dd740637693dbb6723e8730f32da43696/lib/chef/resource/mount.rb#L34-L35?

Yes, exactly.

**Comment by t2y** [comment]
This is the source code at those days when I sent this PR.

* https://github.com/chef/chef/blob/937faa7e42c6df03b1deec0e8bbc785b7ba33e90/lib/chef/resource/mount.rb#L45

**Comment by t2y** [comment]
@burtlo @kagarmoe The review of my changes has finished. I think there's no task for me. Is it right?

**Comment by kagarmoe** [comment]
gif-keyboard-4234956650618166531

**Pull request merged**